migrations/1602693850245-search_index.js

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-await-in-loop */
+const mongoose = require('mongoose');
+// eslint-disable-next-line no-unused-vars
+const { info, error } = require('../src/logger');
+
+const { connect, close } = require('../src/utils/database');
+
+// use your own name for your model, otherwise other migrations may fail.
+// The third parameter is the actually relevent one for what collection to write to.
+const User = mongoose.model(
+	'makeMeUnique',
+	new mongoose.Schema(
+		{
+			firstName: { type: String, required: true },
+			lastName: { type: String, required: true },
+			email: { type: String, required: true, lowercase: true },
+			searchIndexes: { type: mongoose.Schema.Types.Array },
+		},
+		{
+			timestamps: true,
+		}
+	),
+	'users'
+);
+
+const splitForSearchIndexes = (...searchTexts) => {
+	const arr = [];
+	searchTexts.forEach((item) => {
+		item.split(' ').forEach((it) => {
+			// eslint-disable-next-line no-plusplus
+			if (it.length === 0) return;
+
+			arr.push(it.slice(0, 1));
+			if (it.length > 1) arr.push(it.slice(0, 2));
+			for (let i = 0; i < it.length - 2; i += 1) arr.push(it.slice(i, i + 3));
+		});
+	});
+	return arr;
+};
+
+// How to use more than one schema per collection on mongodb
+// https://stackoverflow.com/questions/14453864/use-more-than-one-schema-per-collection-on-mongodb
+
+// TODO npm run migration-persist and remove this line
+// TODO update seed data and remove this line
+
+module.exports = {
+	up: async function up() {
+		await connect();
+
+		const amount = await User.countDocuments();
+		info(`${amount} users will be udated`);
+		const limit = 500;
+		let skip = 0;
+		let looped = 0;
+
+		while (looped < amount) {
+			// ////////////////////////////////////////////////////
+			// Make changes to the database here.
+			// Hint: Access models via this('modelName'), not an imported model to have
+			// access to the correct database connection. Otherwise Mongoose calls never return.
+			info('load current amount users');
+			const users = await User.find()
+				.select(['firstName', 'lastName', 'email'])
+				.sort({
+					_id: 1,
+				})
+				.skip(skip)
+				.limit(limit)
+				.exec();
+			looped += users.length;
+			skip = looped;
+			info('add index to user');
+			try {
+				await Promise.all(
+					users.map((user) => {
+						user.searchIndexes = splitForSearchIndexes(user.firstName, user.lastName, user.email);
+						return user.save();
+					})
+				);
```

Comment:
Is it a good idea to do that with 500 users in parallel? Also, why .map and not .forEach?

Reply:
reduced to 250, shouldn't be to much, otherwise it takes longer